app/models/guest_macro_parser.rb

```diff
@@ -11,9 +11,9 @@ def parse!
     bmr, age, tdee_coef, weight_rate_factor = 0, 0, 0, 0
 
     if params[:gender] == "M"
-      bmr = ((9.99 * params[:weight].to_f) + (6.25 * params[:height].to_f) - (4.92 * age.to_f) + 5).to_i
+      bmr = ((9.99 * params[:weight].to_f) + (6.25 * params[:height].to_f) - (4.92 * params[:age].to_f) + 5).to_i
```

Comment:
What does these numbers mean? could you comment on that?

Also, the bmr calculation for Male and Female are only different by 1 field (the +5 and the -161), please merge this into 1 line instead where you setup that number as a variable. i.e:

```ruby
class GuestMacroParser
FEMALE = "F"
MALE = "M"
....

def get_bmr
  gender_constants = {FEMALE => -161, MALE => 5}
  gender  = gender_constants[@params[:gender]]

  weight_coef = (9.99 * params[:weight].to_f)
  height_coef = (6.25 * params[:height].to_f)
  age_coef = (4.92 * params[:age].to_f)

  return (weight + height - age + gender).to_i
end
``` 

Notice this is not tested so it's just as an example.

Now, getting the expected bmr coefficient is very easy to test, and you should do so.